### PR TITLE
Librarify

### PR DESCRIPTION
**Pull request by petervdonovan**
This PR makes it possible to use parts of the Lingo code base in WebAssembly.

This PR is very experimental. It is also quite small, so the stakes are low here, and there isn't a big rush to get this merged.

The changes here are sufficient to a) compile this to WASM without errors, and b) make it easy to get the JSON build properties for a given LF file from JS/TS.

More changes would be required in order to make all of the functionality of Lingo work without panics; this is because apparently the norm in the Rust ecosystem is to panic at runtime when encountering an operation that doesn't support WASM instead of using Cargo features, which would result in compile-time errors.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [package]
-name = "lingua-franca"
+name = "lingo"
```
Comment (tanneberger) [inline Cargo.toml:2]: I would leave it as lingua-franca because we still want to squat that name on crates.io ;D

File: src/main.rs [hunk 2]
```diff
@@ -1,35 +1,53 @@
...
     // If it exists the returned path is absolute.
-    let lingo_path = util::find_toml(&env::current_dir().unwrap());
+    let lingo_path = liblingo::util::find_toml(&env::current_dir().unwrap());
 
     // tries to read Lingo.toml
     let mut wrapped_config = lingo_path.as_ref().and_then(|path| {
-        ConfigFile::from(path)
-            .map_err(|err| println!("Error while reading Lingo.toml: {}", err))
+        ConfigFile::from(path, Box::new(do_read_to_string))
+            .map_err(|err| log::error!("Error while reading Lingo.toml: {}", err))
```
Comment (tanneberger) [inline src/main.rs:50]: if we now want to switch to a proper logging library, we probably need some non-standard schema.

File: src/args.rs [hunk 3]
```diff
@@ -7,6 +7,7 @@ use std::path::PathBuf;
 #[clap(rename_all = "lowercase")]
 pub enum TargetLanguage {
     C,
+    CCpp,
```
Comment (tanneberger) [inline src/args.rs:10]: ?

**Comment by tanneberger** [comment]
we probably should setup a github action which checks if lingo still can be compiled into wsm

File: src/args.rs (hunk 3 above)
Comment (lhstrh) [reply src/args.rs:10]: ??

File: Cargo.toml (hunk 1 above)
Comment (lhstrh) [reply Cargo.toml:2]:
```suggestion
name = "lingua-franca"
```

File: src/main.rs (hunk 2 above)
Comment (lhstrh) [reply src/main.rs:50]: @tanneberger, is there anything concrete and actionable here?

**Review by lhstrh — APPROVED** [review]
:+1:

**Pull request merged**